model/driver/src/icon4py/model/driver/initialization_utils.py

```diff
@@ -428,6 +437,8 @@ def read_static_fields(
      Args:
         grid: IconGrid
```

Comment:
If you have the `grid` here already as an argument isnt the GridShape and the uuid available from it in the IconGrid.global_properties`?